json/src/de.rs

```diff
@@ -828,3 +829,53 @@ pub fn from_str<T>(s: &str) -> Result<T>
 {
     from_slice(s.as_bytes())
 }
+
+/// Iterator over JSON values
+pub struct JSONStream<T, Iter>
+    where Iter: Iterator<Item=io::Result<u8>>,
+          T: de::Deserialize
+{
+    deser: Deserializer<Iter>,
+    _marker: PhantomData<T>,
+}
+
+impl <T, Iter>JSONStream<T, Iter>
+    where Iter:Iterator<Item=io::Result<u8>>,
+          T: de::Deserialize {
+    fn new(i: Iter) -> JSONStream<T, Iter> {
```

Comment:
Yes please just change the `new` method to be public. The `parse_stream` doesn't really add anything here, so it'd be easier to have users use the `new` constructor.